### PR TITLE
First implementation of Mod Settings UI

### PR DESCRIPTION
**Pull request by paulhobbel**
The first Mod Settings UI implementation

General settings:
image

Mod settings:
image

It still needs some work to receive active mods from VU itself and show only those while ingame. The ingame menu does work though and has the same behaviour as the current settings. Only audio an amd voip settings are shown

### CODE REVIEW

File: app/components/inputs/OptionsInput.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        }
+
+        return (
+            <Select
+                isSearchable={false}
+                styles={SELECT_STYLE}
+                value={val}
+                options={options}
+                onChange={this.props.onChange}
+                isClearable={this.props.allowEmpty??false}
```
Comment (OrfeasZ) [inline app/components/inputs/OptionsInput.js]: This can probably be simplified as `!!this.props.allowEmpty`.

File: app/containers/Settings.js [hunk 2]
```diff
@@ -106,113 +57,228 @@ class Settings extends Component
...
 
         if (voipDevices.length === 0)
         {
             voipDevices.push({ value: -1, label: 'No microphone detected' });
-            selectedDevice = 0;
+            selectedDevice = 0; // I don't think we need this one
         }
 
-        let gameSettings = (
+        let audioSettings = (
```
Comment (OrfeasZ) [inline app/containers/Settings.js]: Might be a good idea to move these into their own components (audio, game, mods).

File: app/containers/Settings.js [hunk 3]
```diff
@@ -106,113 +57,228 @@ class Settings extends Component
...
+                    />;
+                case KEYBIND:
+                    return <KeybindInput
+                        value={setting.currentValue !== undefined ? setting.currentValue : setting.value}
+                        onChange={(e) => {
+                            this._onChangeModInput(settingKey, e);
+                        }}
+                    />;
+                case MULTI_KEYBIND:
+                    return  <>Unsupported</>; // TODO: Fixme
```
Comment (OrfeasZ) [inline app/containers/Settings.js]: This should be implemented prior to merging.

File: assets/scss/_settings.scss [hunk 4]
```diff
@@ -1,79 +1,174 @@
...
+      }
+    }
+  }
+
+  .tab-inner {
+    position: relative;
+    border: 0.15741vh solid rgba(255, 255, 255, 0.2);
+    height: calc(100% - 11.703703703703704vh);
+    background: rgba(0, 0, 0, 0.8);
+    backdrop-filter: blur(10px);
```
Comment (OrfeasZ) [inline assets/scss/_settings.scss]: Pixel value should probably be vh.

File: app/constants/Styles.js [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+            color,
+            fontWeight,
+            fontSize: '1.851851851851852vh', // 20px
+            padding: '0.7407407407407407vh 1.111111111111111vh', // 8px 12px,
+        };
+    },
+    menu: (provided) => ({
+        ...provided,
+        backgroundColor: 'rgba(0, 0, 0, 0.9)',
+        backdropFilter: 'blur(20px)',
```
Comment (OrfeasZ) [inline app/constants/Styles.js]: Px value should probably be in vh.

**Review by OrfeasZ — CHANGES_REQUESTED** [review]

File: app/components/settings/AudioSettings.js [hunk 6]
```diff
@@ -0,0 +1,115 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import Select from 'react-select';
+
+import * as ActionTypes from "../../constants/ActionTypes";
+import { SELECT_STYLE } from "../../constants/Styles";
+import VoipSlider from "../VoipSlider";
+import Slider from "../Slider";
+
+class AudioSettings extends Component
```
Comment (paulhobbel) [inline app/components/settings/AudioSettings.js]: This is a container or smart component so it should be in the containers folder. Or you have to make it a dumb component and pass changes by events

**Review by OrfeasZ — APPROVED** [review]

**Pull request merged**